test/cpp/common/alarm_test.cc

```diff
@@ -266,6 +266,30 @@ TEST(AlarmTest, NegativeExpiry) {
   EXPECT_EQ(junk, output_tag);
 }
 
+// Infinite past or unix epoch should fire immediately.
+TEST(AlarmTest, InfPastExpiry) {
+  CompletionQueue cq;
+  void* junk = reinterpret_cast<void*>(1618033);
+  Alarm alarm;
+  alarm.Set(&cq, gpr_inf_past(GPR_CLOCK_REALTIME), junk);
+
+  void* output_tag;
+  bool ok;
+  CompletionQueue::NextStatus status =
+      cq.AsyncNext(&output_tag, &ok, grpc_timeout_seconds_to_deadline(1));
```

Comment:
On a heavily loaded system with low resources, or maybe running this 10,000 times, I do wonder if we'll see flakes due to the short deadline. It may be safer to extend the deadline to 10s, given we expect it to complete immediately regardless.

Reply:
Makes sense. Done.